src/Components/UnlistedArtworkBanner.tsx

```diff
@@ -0,0 +1,47 @@
+import { createFragmentContainer, graphql } from "react-relay"
+import { Text, Clickable } from "@artsy/palette"
+import { useState } from "react"
+import { FullBleedBanner } from "Components/FullBleedBanner"
+import { UnlistedArtworkBanner_partner$data } from "/__generated__/UnlistedArtworkBanner_partner.graphql"
+import { UnlistedBannerModal } from "Apps/Artwork/Components/UnlistedBannerModal"
+
+interface UnlistedArtworkBannerProps {
+  partner?: UnlistedArtworkBanner_partner$data
+}
+
+const UnlistedArtworkBanner: React.FC<UnlistedArtworkBannerProps> = ({
+  partner,
+}) => {
+  const [open, setOpen] = useState(false)
+
+  return (
+    <>
+      <FullBleedBanner variant="brand">
+        <Text data-testid="unlisted-text">
+          This is a{" "}
+          <Clickable
+            textDecoration="underline"
+            onClick={() => {
+              setOpen(true)
+            }}
+          >
+            private listing
+          </Clickable>{" "}
+          {partner?.name ? `from ${partner.name}.` : "."}
+        </Text>
+      </FullBleedBanner>
+      <UnlistedBannerModal show={open} onClose={() => setOpen(false)} />
```

Comment:
Was following the pattern of extracting the modal out into its own component here.